### PR TITLE
Added Hessian inversion error message if Hessian is singular. Fixed s…

### PR DESCRIPTION
**Pull request by nfick**
…mall typo in comment.  This is my first try at something like this.  I haven't figured out how to test the code I added.

### CODE REVIEW

File: statsmodels/discrete/discrete_model.py [hunk 1]
```diff
@@ -373,7 +373,11 @@ def cov_params_func_l1(self, likelihood_model, xopt, retvals):
         if nnz_params > 0:
             H_restricted = H[nz_idx[:, None], nz_idx]
             # Covariance estimate for the nonzero params
-            H_restricted_inv = np.linalg.inv(-H_restricted)
+            try:
+                H_restricted_inv = np.linalg.inv(-H_restricted)
+            except np.linalg.LinAlgError:
+                print 'Hessian inversion failed: Hessian is singular ' \
```
Comment (josef-pkt) [inline statsmodels/discrete/discrete_model.py]: no `print` in code
this should be 
```
import warnings
warnings.warn(...)
```

**Comment by josef-pkt** [comment]
Thanks,

Do you have a simple failing example?

**Comment by nfick** [comment]
No, I should have thought to at least have that.  I think I will have to work on this a good bit more.  I guess I will do another pull request after I figure out how to test, if no one resolves this issue by then.  Sorry, I am completely new to this.

**Comment by josef-pkt** [comment]
For the actual unit test: if you have an example, then it is mainly an `assert_warns` which IIRC we have from numpy.testing (or also from pandas).

I don't remember where we used it, since we don't test warnings frequently enough yet.

**Comment by josef-pkt** [comment]
I can help a bit later, if anything is unclear.

**Comment by josef-pkt** [comment]
This was changed at some point to use the try.. except.
And has been changed again in #3099 to use eigenvalues to check for non-positive definite instead.

**Pull request closed without merging**